### PR TITLE
Add query id to hive connector to build bucket file name

### PR DESCRIPTION
**Pull request by xiaoxmeng**
In Presto e2e test, we found the bucket file name is not compatible
with one generated by Presto java if some bucket file is missing
(computeFileNamesForMissingBuckets). The difference is that java
use query id to build bucket file name but velox use the task id.
We can just use query id which is unique under a particular partition
dir so no need to use task id.

### CODE REVIEW

File: velox/dwio/common/CMakeLists.txt [hunk 1]
```diff
@@ -21,6 +21,8 @@ elseif(${VELOX_BUILD_TEST_UTILS})
   add_subdirectory(tests/utils)
 endif()
 
+include_directories(/opt/homebrew/opt/protobuf/include)
```
Comment (mbasmanova) [inline velox/dwio/common/CMakeLists.txt:24]: Is this needed? Seems unrelated.

File: velox/exec/tests/TableWriteTest.cpp [hunk 2]
```diff
@@ -440,25 +440,6 @@ class TableWriteTest : public HiveConnectorTestBase {
     }
   }
 
-#if 0
```
Comment (mbasmanova) [inline velox/exec/tests/TableWriteTest.cpp:443]: Thanks.

File: velox/exec/tests/TableWriteTest.cpp [hunk 3]
```diff
@@ -475,13 +456,13 @@ class TableWriteTest : public HiveConnectorTestBase {
       const std::string& targetDir) {
     verifyPartitionedDirPath(filePath, targetDir);
     if (commitStrategy_ == CommitStrategy::kNoCommit) {
-      ASSERT_TRUE(RE2::FullMatch(
-          filePath.filename().string(), "0[0-9]+_0_test_cursor.+"))
+      // NOTE: the query id is empty in unit test.
```
Comment (mbasmanova) [inline velox/exec/tests/TableWriteTest.cpp:459]: Can we make it non-empty?

File: velox/dwio/common/CMakeLists.txt (hunk 1 above)
Comment (xiaoxmeng) [reply velox/dwio/common/CMakeLists.txt:24]: It is required locally. Let me remove.

**Review by mbasmanova — APPROVED** [review]

File: velox/exec/tests/TableWriteTest.cpp (hunk 3 above)
Comment (xiaoxmeng) [reply velox/exec/tests/TableWriteTest.cpp:459]: Good point. I thought about and add a query id in task cursor.

**Review by kewang1024 — APPROVED** [review]
NIT: typo in description

We can just use query id as bucket id **is** unique under a particular...

**Comment by xiaoxmeng** [comment]
> NIT: typo in description
> 
> We can just use query id as bucket id **is** unique under a particular...

@kewang1024 thanks!
